jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartFormData.java

```diff
@@ -468,32 +479,41 @@ public void onPartContent(Content.Chunk chunk)
                             onFailure(x);
                         }
 
-                        partChunks.forEach(Content.Chunk::release);
-                        partChunks.clear();
+                        try (AutoLock ignored = lock.lock())
+                        {
+                            partChunks.forEach(Content.Chunk::release);
+                            partChunks.clear();
+                        }
                         return;
                     }
                 }
             }
             // Retain the chunk because it is stored for later use.
             chunk.retain();
-            partChunks.add(chunk);
+            try (AutoLock ignored = lock.lock())
+            {
+                partChunks.add(chunk);
+            }
         }
 
         private void write(ByteBuffer buffer) throws Exception
         {
             int remaining = buffer.remaining();
             while (remaining > 0)
             {
-                int written = fileChannel.write(buffer);
-                if (written == 0)
-                    throw new NonWritableChannelException();
-                remaining -= written;
+                try (AutoLock ignored = lock.lock())
+                {
+                    int written = fileChannel.write(buffer);
+                    if (written == 0)
+                        throw new NonWritableChannelException();
+                    remaining -= written;
+                }
             }
         }
 
         private void close()
         {
-            try
+            try (AutoLock ignored = lock.lock())
```

Comment:
Would not do the close with the lock held.
Can we just have a `fileChannel()` method that locks and returns the field?

Reply:
Same as above.